README.md

```diff
@@ -1,6 +1,394 @@
+
+##### Broker Objects
+
+- *ServiceInstance*
+
+  A Broker's internal representation of a provisioned service.
+
+- *ServiceInstanceCredential*
+
+  A Broker's data structure for tracking coordinates and auth credentials for a single *ServiceInstance*.  
+
+- *Catalog*
+
+  A complete list of services and plans that are provisionable by the Broker.
+
+#### Local Kubernetes Cluster
+
+For this example, the Local Kubernetes Cluster is used to represent a cluster in which active development takes place.
+The Service-Catalog must be run here because it will be making Kubernetes API calls to create API objects for developers.
+
+##### Service-Catalog API Objects
+
+Here is where naming collisions become confusing.
+All of the terms in this section are objects of the Service Catalog API Server.
+The terms here are only Kubernetes' representation of the actual services provisioned in the External Service Provider.
+They are managed by the `SC-APISERVER` portion of the [control flow diagram](docs/diagram/control-diag.md).
+
+**NOTE: There is no Catalog object in the Service Catalog.  It is represented as a set of ServiceClasses only!**
+
+- *ServiceBroker*
+
+  Service Catalog representation of the actual Broker, usually located elsewhere.
+  The Service Catalog Controller Manager will use the URL provided within the *ServiceBroker* to connect to the actual Broker server.
+  A *ServiceBroker* can offer many *ServiceClasses*.
+
+- *ServiceClass*
+
+  Service Catalog representation of a *Catalog* offering.  When a *ServiceBroker* is created, the Service Catalog Controller Manager requests the *Catalog* from the actual Broker.
+  The response is a json object listing all Services and Plans offered by the Broker.
+  The Controller Manager processes this response into a set of *ServiceClasses* in the API Server.
+  In this case, our *ServiceClass* is a provisionable object bucket.
+  A *ServiceClass* can provision many *ServiceInstances* of its class.
+
+- *ServiceInstance*
+
+  Service Catalog representation of a consumable service instance in the External Service Provider.
+  In this case, a single object bucket.
+  A *ServiceInstance* can have many *ServiceInstanceCredentials*, so long as the service supports this.
+  This enables a single instance to be consumed by many Pods.
+
+- *ServiceInstanceCredential*
+
+  Service Catalog object that does not contain any authentication or coordinate information.
+  Instead, when a *ServiceInstanceCredential* is created in the Service Catalog API Server, it triggers a request to the Broker for authentication and coordinate information.
+  Once a response is received, the sensitive information is stored in a *Secret* in the same namespace as the *ServiceInstanceCredential*.
+
+### What It Does
+This broker implements [OpenServiceBroker](https://github.com/openservicebrokerapi/servicebroker/blob/master/spec.md) methods for creating, connecting to, and destroying
+object buckets.
```

Comment:
"S3 compatible buckets"

Reply:
replacing all instances of "object bucket" with "S3 bucket" for clarity.